ergo-p2p/src/peer_addr.rs

```diff
@@ -0,0 +1,90 @@
+//! Peer address types
+use std::{
+    convert::TryInto,
+    net::{IpAddr, Ipv4Addr, SocketAddr, SocketAddrV4},
+};
+
+use derive_more::{Display, From, Into};
+use sigma_ser::{ScorexSerializable, ScorexSerializationError};
+
+/// Peer address
+#[derive(PartialEq, Eq, Debug, Copy, Clone, From, Into, Hash, Display)]
+pub struct PeerAddr(SocketAddr);
+
+impl PeerAddr {
+    /// Size in bytes of the ip address associated with this peer address
+    pub fn ip_size(&self) -> usize {
+        match self.0.ip() {
+            IpAddr::V4(ip) => ip.octets().len(),
+            IpAddr::V6(ip) => ip.octets().len(),
+        }
+    }
+}
+
+impl ScorexSerializable for PeerAddr {
+    fn scorex_serialize<W: sigma_ser::vlq_encode::WriteSigmaVlqExt>(
+        &self,
+        w: &mut W,
+    ) -> sigma_ser::ScorexSerializeResult {
+        let ip = match self.0.ip() {
+            IpAddr::V4(ip) => ip,
+            _ => return Err(ScorexSerializationError::NotSupported("ipv6 not supported")),
+        };
+
+        w.write_all(&ip.octets())?;
+        w.put_u32(self.0.port() as u32)?;
+
+        Ok(())
+    }
+
+    fn scorex_parse<R: sigma_ser::vlq_encode::ReadSigmaVlqExt>(
+        r: &mut R,
+    ) -> Result<Self, sigma_ser::ScorexParsingError> {
+        let mut fa = [0u8; 4];
+        r.read_exact(&mut fa)?;
+
+        let ip = Ipv4Addr::from(fa);
+        let port: u16 = r.get_u32()?.try_into()?;
+
+        Ok(SocketAddr::V4(SocketAddrV4::new(ip, port)).into())
+    }
+}
+
+/// Arbitrary
+#[cfg(feature = "arbitrary")]
+pub mod arbitrary {
+    use super::*;
+    use proptest::prelude::*;
+    use proptest::prelude::{Arbitrary, BoxedStrategy};
+
+    impl Arbitrary for PeerAddr {
+        type Parameters = ();
+        type Strategy = BoxedStrategy<Self>;
+
+        fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+            (any::<[u8; 4]>(), any::<u16>())
+                .prop_map(|(octets, port)| {
+                    SocketAddr::new(Ipv4Addr::from(octets).into(), port).into()
+                })
+                .boxed()
+        }
+    }
+}
+
```

Comment:
```suggestion

#[cfg(feature = "arbitrary")]
```
Should help. Or, alternatively maybe adding `proptest` to `dev-dependencies` will fix it as well.

Reply:
Neither done the trick, here's what I think is happening:

When running tests in a workspace if one dependency enables a feature then it's enabled across all of them for all their tests:
- cargo test --package ergotree-ir fails
- cargo test passes because ergo-lib enables the arbitrary feature: ergotree-ir = { version = "^0.14.0", path = "../ergotree-ir", features = ["arbitrary"] }

Tests are passing now because they're behind the `arbitrary` feature, which I think makes sense because the tests are completely dependent on `Arbitrary`. I think the way to go is adding `--features=arbitrary` to the ci action?

Reply:
Thanks for digging into this! I tested and `ergotree-ir` crate behaves the same way. I'm not a fan of adding `--features=arbitrary`. I'd prefer tests run without any additional options. I found this workaround - https://github.com/rust-lang/cargo/issues/2911#issuecomment-749580481 What do you think?

Reply:
I agree with not needing to enable additional options for tests, added the workaround 👍